subsys/shell/backends/shell_telnet.c

```diff
@@ -85,12 +85,31 @@ static void telnet_reply_ay_command(void)
 	telnet_command_send_reply((uint8_t *)alive, strlen(alive));
 }
 
+static int telnet_echo_set(const struct shell *sh, bool val)
+{
+	int ret = shell_echo_set(sh_telnet->shell_context, val);
+
+	if (ret < 0) {
+		LOG_ERR("Failed to disable echo, err: %d", ret);
+	}
+	return ret;
+}
+
 static void telnet_reply_do_command(struct telnet_simple_command *cmd)
 {
 	switch (cmd->opt) {
 	case NVT_OPT_SUPR_GA:
 		cmd->op = NVT_CMD_WILL;
 		break;
+	case NVT_OPT_ECHO:
+		int ret = telnet_echo_set(sh_telnet->shell_context, true);
+
+		if (ret >= 0) {
+			cmd->op = NVT_CMD_WILL;
+		} else {
+			cmd->op = NVT_CMD_WONT;
+		}
+		break;
```

Comment:
Question: What's your guess with regard to how likely it is that upstream will want this as a {{Kconfig}} option? I suppose it still might be better not to do so until they ask for it...

Reply:
Actually, since the consequences on traffic are pretty high, and informed by [this comment on their commit](https://github.com/zephyrproject-rtos/zephyr/commit/51dc4c3b1eff31a04e7e9121f5cd9a1dac1789e7), I'm gonna guess that they'll want something like `SHELL_TELNET_SUPPORT_COMMAND_ECHO`.